src/matter/cluster/ScenesCluster.ts

```diff
@@ -0,0 +1,293 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    ArrayT,
+    BooleanT,
+    Field,
+    OptionalField,
+    ObjectT,
+    StringT,
+    UInt8T,
+    UInt16T,
+    UInt32T,
+    Bound,
+    EnumT, BitMapT, Bit
+} from "../../codec/TlvObjectCodec";
+import { Attribute, OptionalAttribute, Cluster, Command, OptionalCommand, NoResponseT } from "./Cluster";
+import { StatusCode } from "../interaction/InteractionMessages";
+import { GroupIdT } from "../common/GroupId";
+import { ClusterIdT } from "../common/ClusterId";
+import { NodeIdT } from "../common/NodeId";
+import { AttributeIdT } from "../common/AttributeId";
+import { MatterApplicationClusterSpecificationV1_0 } from "../../Specifications";
+
+/** @see {@link MatterApplicationClusterSpecificationV1_0} § 1.4.6.1 */
+const AttributeValuePairT = ObjectT({
+    /**
+     * This field SHALL be present or not present, for all instances in the Scenes cluster. If this field is
```

Comment:
:-) This field SHALL be present or not present : difficult not to respect this rule, I think all instances in the Scenes cluster will respect it.

Reply:
I removed the "optional" for now ... because if fieldname is optional the the whole entry makes no sense.